src/raster/r.in.ahn/r.in.ahn.py

```diff
@@ -0,0 +1,181 @@
+#               (Actueel Hoogtebestand Nederland (AHN), version 4.
+#
+# COPYRIGHT:    (c) 2024 Paulo van Breugel, and the GRASS Development Team
+#               This program is free software under the GNU General Public
+#               License (>=v2). Read the file COPYING that comes with GRASS
+#               for details.
+#
+#############################################################################
+
+# %module
+# % description: Imports the 0.5 meter resolution dtm or dsm from the AHN (Actueel Hoogtebestand Nederland (AHN), version 4.
+# % keyword: dem
+# % keyword: raster
+# % keyword: import
+# %end
+
+# %option
+# % key: product
+# % type: string
+# % label: product
+# % description: Choose which product to download (dsm or dtm)
+# % options: dsm, dtm
+# % required: yes
+# %end
+
+# %option G_OPT_R_OUTPUT
+# % key_desc: File name
+# % label: Name of output raster layer
+# %end
+
+# %flag
+# % key: t
+# % label: Download whole tiles
+# % description: Download the selected product for the AHN tile that overlap with the current region. If enabled, a vector layer with the bounds of the AHN tiles will be saved, using the output name with as suffix _tiles
+# %end
+
+# %flag
+# % key: g
+# % label: Set to computational region
+# % description: After downloading and importing the requested layer, set the region back to the original computation region.
+# %end
+
+
+import atexit
+import sys
+import grass.script as gs
+from math import floor
+import uuid
+
+# create set to store names of temporary maps to be deleted upon exit
+CLEAN_LAY = []
+
+
+def cleanup():
+    """Remove temporary maps specified in the global list"""
+    maps = reversed(CLEAN_LAY)
+    mapset = gs.gisenv()["MAPSET"]
+    for map_name in maps:
+        for element in ("raster", "vector", "region"):
+            found = gs.find_file(name=map_name, element=element, mapset=mapset)
+            if found["file"]:
+                gs.run_command(
+                    "g.remove", flags="f", type=element, name=map_name, quiet=True
+                )
+
+
+def create_temporary_name(prefix):
+    tmpf = f"{prefix}{str(uuid.uuid4().hex)}"
+    CLEAN_LAY.append(tmpf)
+    return tmpf
+
+
+def main(options, flags):
+    """
+    Download the dtm or dsm from AHN
+    """
+
+    # Check if the projection is RD New (EPSG:28992)
+    proj_info = gs.parse_command("g.proj", flags="g")
+    if proj_info["srid"] != "EPSG:28992" and proj_info["name"] != "Amersfoort / RD New":
+        gs.fatal(_("This module only works with locations with projection EPSG=28992"))
+
+    # Check if wcs addon is installed.
+    try:
+        gs.find_program("r.in.wcs")
+    except Exception as e:
+        gs.fatal(_("The addon r.in.wcs is not installed. Exit"))
+
+    # Get coordinates current region extent
+    if flags["g"]:
+        region_original = create_temporary_name("region_")
+        gs.run_command("g.region", save=region_original)
+    region_current = gs.parse_command("g.region", flags="gu")
```

Comment:
You should probably use `gs.use_temp_region` and `gs.del_temp_region` in cleanup.

Reply:
done